### ISSUE TITLE
Broken deployments on Windows - packrat/rsconnect conflict

### ISSUE THREAD

**Issue by cwthom** [rstudio/rsconnect#369]
Hello,

Since upgrading to v0.8.15, deployments from my Windows development environments to RStudio Connect have started failing. The error message is:
```
arguments imply differing number of rows: 1, 0
```
which I've tracked down to `rsconnect:::snapshotDependencies`. Specifically, I think the code introduced in #309 is the problem for Windows users - the use of `contrib.url` in
```
repo.packages <- available.packages(contriburl = contrib.url(repos, type = "source"),
                                    type = "source")
```
and
```
repo.lookup <- data.frame(name = names(named.repos), 
                          url = as.character(named.repos), 
                          contrib.url = contrib.url(named.repos), 
                          stringsAsFactors = FALSE)
```
The first instance of `contrib.url` specifies `type = "source"`, but the second uses platform default (i.e. win-binary). This means that in:
```
package.repo <- repo.lookup[repo.lookup$contrib.url == package.contrib, ]
```
there won't be a match on Windows, and `snapshotDependencies` errors.

Let me know if you need more details on this, happy to provide. I don't know if setting `type = "source"` in the second instance as well would fix this?

Many thanks as always for all your work!
Chris

**Comment by jmcphers** [comment]
Thank you for the detailed analysis! We'll investigate. Any chance you can share the package list for the app that's causing this trouble? (we couldn't reproduce when publishing a simple Shiny app)

**Comment by cwthom** [comment]
Hi @jmcphers - I've dug into this a bit further, and can get it to occur with a very simple app:
```
library(shiny)
library(shinydashboard)
library(rsconnect)

ui <- dashboardPage(
  
  header = dashboardHeader(),
  sidebar = dashboardSidebar(),
  body = dashboardBody()
  
)

server <- function(input, output, session) {}

shinyApp(ui, server)
```

However, the error only occurs given 2 conditions:
- packrat is enabled in the project
- the deployment is done manually with `deployApp`, not push-button deployment

On closer inspection this is a packrat issue not an rsconnect issue. Hadn't realised before but the deployments which started failing were all from packrat-enabled projects. This is because in `packrat::on`, digging down you get to:
```
packrat:::ensurePkgTypeNotBoth
function () 
{
    oldPkgType <- getOption("pkgType")
    if (identical(oldPkgType, "both")) 
        options(pkgType = .Platform$pkgType)
    oldPkgType
}
```
So the issue seems to be that packrat is ensuring that the "pkgType" option is not "both", whereas on Windows (and I guess Mac?), rsconnect relies on it being "both". 

It feels like this conflict would be easier to resolve by changing rsconnect rather than by changing packrat, but I don't fully understand the implications in either case (or understand the difference between `deployApp` and the push-button method in the IDE).

**Comment by cwthom** [comment]
On reflection I guess the IDE just comes bundled with versions of packages like rsconnect and knitr that are exposed as buttons and live in a hidden library somewhere, so I probably just have a slightly older version of the IDE?

**Comment by kevinushey** [comment]
Fixed via #370.

**Comment by cwthom** [comment]
Thanks @kevinushey and @jmcphers !
